src/Expr.js

```diff
@@ -169,35 +228,114 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
+
+  if ('containsstr' in expr) {
+    return (
+      'ContainsStr(' +
+      exprToString(expr['containsstr']) +
+      ', ' +
+      exprToString(expr['search']) +
+      ')'
+    )
+  }
+
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    var argumentExpr = exprToString(expr['arguments'])
+
+    // Print when arguments are an object
+    if (
+      typeof expr['arguments'] === 'object' &&
+      !Array.isArray(expr['arguments']) &&
+      !expr['arguments']['raw']
+    ) {
+      argumentExpr = printObject(expr['arguments'])
+    }
+
+    return 'Call(' + exprToString(expr['call']) + ', ' + argumentExpr + ')'
+  }
+
   // Versioned queries/lambdas will have an api_version field.
-  // We want to prevent it from being parsed and displayed as:
-  // Query(ApiVersion("3", "X", Var("X")))
+  // We want to prevent it from being parsed and displayed.
   var keys = Object.keys(expr).filter(
     expression => expression !== 'api_version'
```

Comment:
given lambdas are already handled before this point, that filter is not necessary